x-pack/elastic-agent/pkg/agent/application/pipeline/dispatcher/dispatcher_test.go

```diff
@@ -49,8 +49,9 @@ func TestActionDispatcher(t *testing.T) {
 	ack := noopacker.NewAcker()
 
 	t.Run("Success to dispatch multiples events", func(t *testing.T) {
+		ctx := context.Background()
```

Comment:
Given [my change proposal](https://github.com/elastic/beats/pull/30385/files#r813779788), better to add a test.


This is the type definition, I couldn't add as a change suggestion
```
type mockAcker struct {
	CommitFn func(ctx context.Context) error
}

func (m mockAcker) Ack(ctx context.Context, action fleetapi.Action) error {
	panic("implement me")
}

func (m mockAcker) Commit(ctx context.Context) error {
	return m.CommitFn(ctx)
}
```

```suggestion
	ack := noopacker.NewAcker()

	t.Run("Merges ActionDispatcher ctx cancel and Dispatch cxt value", func(t *testing.T) {
		action1 := &mockAction{}
		def := &mockHandler{}

		span := apmtest.NewRecordingTracer().
			StartTransaction("ignore", "ignore").
			StartSpan("ignore", "ignore", nil)

		ctx1, cancel := context.WithCancel(context.Background())
		ack := mockAcker{CommitFn: func(ctx context.Context) error {
			// ctx1 not cancelled yet
			require.NoError(t, ctx.Err())

			got := apm.SpanFromContext(ctx)
			require.Equal(t, span.TraceContext().Span, got.ParentID())

			cancel() // cancel function from ctx1
			require.Error(t, ctx.Err(), context.Canceled)

			return nil
		}}

		d, err := New(ctx1, nil, def)
		require.NoError(t, err)

		ctx2 := apm.ContextWithSpan(context.Background(), span)
		err = d.Dispatch(ctx2, ack, action1)
		require.NoError(t, err)
	})

	t.Run("Success to dispatch multiples events", func(t *testing.T) {
		ctx := context.Background()
```